src/display/transport_stream.js

```diff
@@ -62,17 +76,20 @@ class PDFDataTransportStream {
     this._pdfDataRangeTransport.transportReady();
   }
 
-  _onReceiveData(args) {
-    const buffer = new Uint8Array(args.chunk).buffer;
-    if (args.begin === undefined) {
+  _onReceiveData({ begin, chunk }) {
```

Comment:
The chunk can be an `ArrayBuffer`: 
https://searchfox.org/mozilla-central/rev/a156a65ced2dae5913ae35a68e9445b8ee7ca457/toolkit/components/pdfjs/content/PdfJsNetwork.jsm#184
it's used in `onDone`:
https://searchfox.org/mozilla-central/rev/a156a65ced2dae5913ae35a68e9445b8ee7ca457/toolkit/components/pdfjs/content/PdfStreamConverter.jsm#717-725
and then:
https://github.com/mozilla/pdf.js/blob/d6f63b5fb7cc63fa5d38ff71a56b8ee46d4f1b33/src/display/transport_stream.js#L47

You can saw that with:
https://bibliotekacyfrowa.ujk.edu.pl//Content/4430/PDF/475183.pdf

We should really have consistent types everywhere...

Reply:
It seems that the code in `PdfStreamConverter.jsm` consistently, at least I think so, returns data in as a `Uint8Array`. Hence we might want to simply change the `PdfJsNetwork.jsm` code to be consistent with that?

Perhaps replacing the existing `getArrayBuffer` with something like:
```javascript
function getTypedArray(xhr) {
  const data = xhr.response;
  if (typeof data !== "string") {
    return new Uint8Array(data);
  }
  const length = data.length,
    array = new Uint8Array(length);
  for (let i = 0; i < length; i++) {
    array[i] = data.charCodeAt(i) & 0xff;
  }
  return array;
}
```
and update [this line](https://searchfox.org/mozilla-central/rev/a156a65ced2dae5913ae35a68e9445b8ee7ca457/toolkit/components/pdfjs/content/PdfJsNetwork.jsm#184) accordingly could work?

If so we should probably do that *first*, since the existing code in `src/display/transport_stream.js` should handle that (unless I'm missing something).

Reply:
I wrote the patch and it works with the pdf in bug 1809164.
Do you already have a wip you'll submit ? or should I submit mine ?